### ISSUE TITLE
Wrong classname when upload images in frontend

### ISSUE THREAD

**Issue by MarcioBarrientos** [unclecheese/silverstripe-dropzone#6]
When I upload an image in the frontend the classname for the saved image is set as "file" in the database
In the modeladmin the image is saved as "image" in the database (as expected)

// Dataobject
 private static $has_one = array(
        'Picture' => 'Image'
    );

// Form controller
FileAttachmentField::create('Picture', 'Picture')->setView('grid')

**Comment by unclecheese** [comment]
I can't replicate this. 

You're probably not binding a dataobject to the form using `loadDataFrom()`. Otherwise the `upload()` method has no idea about the file relationship it's saving into, so it just creates a plain `File`. ModelAdmin forms always use `loadDataFrom($record)` because they're used for editing records.

**Comment by anselmdk** [comment]
I'm having the same issue using this on a custom backend without any relation (just using it for uploading images to a folder).
The saved images are given the class name `File`.
I'll probably build a workaround for now. 
Pointer: `$className = File::get_class_for_file_extension(pathinfo($name, PATHINFO_EXTENSION));`.

**Comment by anselmdk** [comment]
This solved the issue for me so far - I'm looping through the folder I'm uploading to, and:

		foreach ($folder->myChildren() as $child) {
			$className = File::get_class_for_file_extension(pathinfo($child->Filename, PATHINFO_EXTENSION));
			if ($child->ClassName != $className) {
				$child->ClassName = $className;
				$child->write();
			}
		}

I know it's a bit of a hack...

**Comment by unclecheese** [comment]
Could you check that changing `FileAttachmentField::getFileClass()` (line 735) to the following fixes the issue?

```php
    public function getFileClass($filename = null) {        
        $name = $this->getName();
        $record = $this->getRecord();
        $ext = pathinfo($filename, PATHINFO_EXTENSION);
        $defaultClass = File::get_class_for_file_extension($ext);            

        if(empty($name) || empty($record)) {
            return $defaultClass;
        }

        $class = $record->getRelationClass($name);
        if(!$class) $class = "File";

        if($filename) {
            if($defaultClass == "Image" && 
               $this->config()->upgrade_images && 
               !Injector::inst()->get($class) instanceof Image
            ) {                
                $class = "Image";
            }
        } 

        return $class;       
    }
```

**Comment by anselmdk** [comment]
Yes, that seems to fix the issue!

**Comment by unclecheese** [comment]
Resolved in 9c547505ca9d777508f322ec29f168fe6edf8afd
